apps/namadillo/src/App/Ibc/ShieldAllAssetList.tsx

```diff
@@ -0,0 +1,65 @@
+import { Asset } from "@chain-registry/types";
+import { Checkbox } from "@namada/components";
+import { TokenCurrency } from "App/Common/TokenCurrency";
+import { AssetWithBalance } from "atoms/integrations";
+import BigNumber from "bignumber.js";
+import clsx from "clsx";
+import { getAssetImageUrl } from "integrations/utils";
+import { toDisplayAmount } from "utils";
+
+type AssetWithBalanceMap = Record<string, AssetWithBalance>;
+
+type ShieldAllAssetListProps<T extends AssetWithBalanceMap> = {
```

Comment:
I think that in practice we won't get stronger typechecking by using `T extends AssetWithBalanceMap`, because we don't know the assets at compile time, so `T` will always be `AssetWithBalanceMap` instead of something more specific.

If you did want to ensure there is a `checked` entry for each asset, maybe something like dropping the `checked` prop and making `assets` be `(AssetWithBalance & { checked: boolean })[]` could work.

Reply:
Yeah, I think it will also improve readability. Made some changes on this! Thank you :)